benchmark/compute_accuracy.py

```diff
@@ -0,0 +1,182 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2015--, The WGS-HGT Development Team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+# ----------------------------------------------------------------------------
+
+"""
+Given known HGTs, compute precision, recall and F-score for various tools
+=========================================================================
+"""
+
+import sys
+import click
+import re
+
+
+def parse_expected_transfers(ground_truth_f):
+    """ Parse ALF's log file.
+
+    Parameters
+    ----------
+    ground_truth_f: string
+        file descriptor to ALF's log file
+
+    Returns
+    -------
+    expected_transfers: list of tuples
+        list of transfers with each tuple representing (organism donor, gene
+        donated, organism recipient, gene received)
+
+    Notes
+    -----
+    Report horizontal gene transfers with information on organism donor,
+    gene donated, organism recipient and gene received.
+    """
+    expected_transfers = []
+    string = "lgt from organism "
+    for line in ground_truth_f:
+        if string in line:
+            content = re.split(
+                'lgt from organism | with gene | to organism |, now gene ',
+                line.strip())
+            expected_transfers.append(tuple(content[1:]))
+    return expected_transfers
+
+
+def parse_observed_transfers(observed_hgts_f,
+                             pvalue_cutoff):
+    """ Parse summary file of observed transfers for various tools.
+
+    Parameters
+    ----------
+    observed_hgts_f: string
+        file descriptor of observed transfers (output of launch_software.sh)
+    pvalue_cutoff: float
+        p-value threshold below which to consider two tree topologies
+        incongruent (incongruence attributed to HGT)
+
+    Returns
+    -------
+    observed_transfers: dict
+        dictionary of tools' names (keys) and a list of horizontal gene
+        transfers
+
+    Notes
+    -----
+    Parse output of launch_software.sh. The outpur file is expected to
+    follow the format:
+
+    #number of HGTs detected
+    #	gene ID	T-REX	RANGER-DTL	RIATA-HGT	Jane 4	Consel
+    0	1000	1	1	1	1	0.96 0.04
+    1	1001	0	0	0	0	0.00 0.00
+    ..
+    """
+    tools = {}
+    tools_id = []
+    next(observed_hgts_f)
+    for line in observed_hgts_f:
+        line = line.strip().split('\t')
+        if line[0].startswith('#'):
+            for tool in line[2:]:
+                tools_id.append(tool)
+                if tool not in tools:
```

Comment:
this pattern could be reduced by replacing `tools = {}` with `tools = defaultdict(list)`. You can in get `defaultdict` from `collections`